sys/vm/vm_map.c

```diff
@@ -3340,6 +3340,7 @@ vm_map_protect(vm_map_t map, vm_offset_t start, vm_offset_t end,
 		if ((flags & VM_MAP_PROTECT_KEEP_CAP) != 0 &&
 		    (entry->max_protection & VM_PROT_CAP) != 0) {
 			new_prot = VM_PROT_ADD_CAP(new_prot);
+			new_maxprot = VM_PROT_ADD_CAP(new_maxprot);
```

Comment:
Do we want a test for this?  Sounds like you could do a RW mmap, store a capability, downgrade to read-only and then see if you can read the tagged capability?

Reply:
I've added a test to do this and verify that store is forbidden.